3-async-await/solution.js

```diff
@@ -19,17 +19,50 @@ Example:
 8. add any needed adjustment to solution() function
 9. as extra challenge: add Promise.race() and Promise.any(), and try to get the idea of what happens
 */
-
-function solution() {
-    // YOUR SOLUTION GOES HERE
-
+const getPrices = require('./prices');
+const getProduct = require('./products');
+async function solution() {
+  try {
     // You generate your id value here
-
+    const id = generateId();
     // You use Promise.all() here
-
-    // You use Promise.allSettled() here
-
-    // Log the results, or errors, here
+    const [price, name] = await Promise.all([getPrices(id), getProduct(id)]);
+    const product = { id, product: name, price };
+    console.log(`\nResolving with Promise.all:\n`, product);
+  } catch (error) {
+    // Log the results, or errors, heres
+    console.log(`\nError in Promise.all: ${error.message}`);
+  }
+}
+async function solutionAllSettled() {
+  const id = generateId();
+  let product = { id };
```

Comment:
I think this product declaration/assignation can be done after the Promise call. You can call the await Promise... first, then if something fails shows the errors.
You can move this part of code before the final console log and create the full object in that part